src/intrinsics/ecma262/Object.js

```diff
@@ -57,7 +60,7 @@ export default function(realm: Realm): NativeFunctionValue {
   });
 
   // ECMA262 19.1.2.1
-  let ObjectAssign = func.defineNativeMethod("assign", 2, (context, [target, ...sources]) => {
+  let ObjectAssignInner = (context, [target, ...sources]) => {
```

Comment:
This should just use a function declaration and a lower case name. The upper case one is a FunctionValue, not a function like the one.